packages/@aws-cdk/assertions/lib/template.ts

```diff
@@ -101,7 +101,7 @@ export class Template {
    * When a literal is provided, performs a partial match via `Match.objectLike()`.
    * Use the `Match` APIs to configure a different behaviour.
    */
-  public findResources(type: string, props: any = {}): { [key: string]: any }[] {
+  public findResources(type: string, props: any = {}): { [key: string]: { [key: string]: any } } {
```

Comment:
oops, I missed this in the last revision. We should be returning `undefined` here when there are no matches, not an empty object.

That would make the user experience better. Users can do -

```ts
const result = template.find...(...);
if (result !== undefined) {
  // do something
}
```

With what we have today, it'll be

```ts
const result = template.find...(...);
if (Object.keys(result).length > 0) {
   // do something
}
```

WDYT?

Reply:
I like this in principle but it will mean that since `result` can be `undefined`, something like `expect(result.Foo).toEqual({})` becomes `expect(result!.Foo).toEqual({})`. I could be wrong but forcing users to use the `!` escape or otherwise dealing with the possibility of `undefined` might not make the user experience better.

WDYT? :)

Reply:
Ok, let's go with what you have now. It might actually be the better way.